pkg/controllers/provisioning/provisioner.go

```diff
@@ -81,34 +81,23 @@ func (p *Provisioner) Add(pod *v1.Pod) <-chan struct{} {
 func (p *Provisioner) provision(ctx context.Context) error {
 	// Batch pods
 	logging.FromContext(ctx).Infof("Waiting for unschedulable pods")
-	items, window := p.batcher.Wait()
```

Comment:
Can easily wait for another PR, but would be nice if batcher didn't collect items and was just a trigger mechanism.